x-pack/packages/security-solution/upselling/pages/entity_analytics.tsx

```diff
@@ -45,15 +45,31 @@ const StyledEuiCard = styled(EuiCard)`
   }
 `;
 
-const EntityAnalyticsUpsellingComponent = () => {
-  const { getAppUrl, navigateTo } = useNavigation();
-  const subscriptionUrl = getAppUrl({
-    appId: 'management',
-    path: 'stack/license_management',
-  });
+const EntityAnalyticsUpsellingComponent = ({
+  requiredLicense,
+  requiredProduct,
+  subscriptionUrl,
+}: {
+  requiredLicense?: string;
+  requiredProduct?: string;
+  subscriptionUrl?: string;
+}) => {
+  const { navigateTo } = useNavigation();
+
   const goToSubscription = useCallback(() => {
     navigateTo({ url: subscriptionUrl });
   }, [navigateTo, subscriptionUrl]);
+
+  if (!requiredProduct && !requiredLicense) {
+    throw new Error('requiredProduct or requiredLicense must be defined');
+  }
+
+  const upgradeMessage = requiredLicense
+    ? i18n.UPGRADE_LICENSE_MESSAGE(requiredLicense)
+    : i18n.UPGRADE_PRODUCT_MESSAGE(requiredProduct ?? '');
+
+  const requiredLicenseOrProduct = requiredProduct ?? requiredLicense ?? '';
```

Comment:
nit, consider renaming since you would get product before license if bother are defined
```suggestion
  const requiredProductOrLicense = requiredProduct ?? requiredLicense ?? '';
```